src/core/service.c

```diff
@@ -2110,7 +2132,9 @@ static void service_enter_start_pre(Service *s) {
         s->control_command = s->exec_command[SERVICE_EXEC_START_PRE];
         if (s->control_command) {
 
-                unit_warn_leftover_processes(UNIT(s));
+                if (unit_warn_leftover_processes(UNIT(s)))
+                        if (service_adverse_to_leftover_processes(s))
+                                goto fail;
```

Comment:
similar here

Reply:
actually, the code in both occasions your new function is called is pretty much the same, hence I figure the invocation of `unit_warn_leftover_processes()` should maybe just move into `service_adverse_to_leftover_process()`? That'd be shorter then, and I like short.